lib/paychex/client/companies.rb

```diff
@@ -12,11 +12,14 @@ def linked_companies
 
         begin
           while response && response.body
-            companies_count = response.body.fetch('metadata').fetch('pagination').fetch('total')
+            companies_count = response.body.fetch('metadata').fetch('pagination').fetch('total') rescue nil

```

Comment:
I feel uncomfortable with `rescue nil`, could we rescue the exception only?

Reply:
Okay, we need to recuse KeyError in order to handle metadata without pagination node